components/brave_rewards/browser/rewards_p3a.cc

```diff
@@ -180,6 +180,63 @@ double CalcWalletBalance(base::flat_map<std::string, double> wallets,
   return balance_minus_grant;
 }
 
+void RecordRewardsEnabledDuration(PrefService* prefs, bool rewards_enabled) {
+  base::Time enabled_timestamp = prefs->GetTime(prefs::kEnabledTimestamp);
+  auto value = RewardsEnabledDuration::kNever;
+
+  if (enabled_timestamp.is_null()) {
+    // No previous timestamp, so record one of the non-duration states.
+    if (!rewards_enabled) {
+      // Rewards have been disabled with no previous timestamp.
+      // Probably they've been on since we started measuring.
+      // Ignore this interval since we can't measure it and treat it
+      // the same as never having enabled.
+      value = RewardsEnabledDuration::kNever;
+    } else {
+      // Rewards have been enabled.
+      // Remember when so we can measure the duration on later changes.
+      prefs->SetTime(prefs::kEnabledTimestamp, base::Time::Now());
+      value = RewardsEnabledDuration::kStillEnabled;
+    }
+  } else {
+    // Previous timestamp available.
+    if (!rewards_enabled) {
+      // Rewards have been disabled. Record the duration they were on.
+      // Set the threshold at three units so each bin represents the
+      // nominal value as an order-of-magnitude: more than three days
+      // is a week, more than three weeks is a month, and so on.
+      constexpr int threshold = 3;
+      constexpr int days_per_week = 7;
+      constexpr double days_per_month = 30.44;  // average length
+      base::TimeDelta duration = base::Time::Now() - enabled_timestamp;
+      DLOG(INFO) << "Rewards disabled after " << duration;
```

Comment:
I'd rather say VLOG, afaik most people don't build the debug browser

Reply:
Ok, I've changed to `VLOG(1)` in 9d97e1cc3ddc2a343040036205bef9f372355cc7.

NB the default developer component build sets `DCHECK_ALWAYS_ON` which I believe [enables DLOG](https://github.com/chromium/chromium/blob/96.0.4664.93/base/logging.h#L526-L551) as well.